frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -209,8 +209,10 @@ function ReaderBookmark:onPosUpdate(pos)
 end
 
 function ReaderBookmark:gotoBookmark(pn_or_xp)
-    local event = self.ui.document.info.has_pages and "GotoPage" or "GotoXPointer"
-    self.ui:handleEvent(Event:new(event, pn_or_xp))
+    if pn_or_xp then
```

Comment:
Is this related to a specific problem you came across?

Reply:
at the first bookmark and previous, or last bookmark and next.
`pn_or_xp` will be nill crashing koreader.
instead of putting a check in every function, i put one check at execution